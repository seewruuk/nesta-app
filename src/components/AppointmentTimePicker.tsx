import { useState, useMemo  } from "react";
import { ReservedAppointment } from "../data/offers";
import {Pressable, ScrollView, View, Text, Platform} from "react-native";
import dayjs from "dayjs";
import 'dayjs/locale/pl';
import localizedFormat from 'dayjs/plugin/localizedFormat';
import DateTimePicker from '@react-native-community/datetimepicker';
import Modal from 'react-native-modal';
import {translation} from "@/src/translation";
import {useStateContext} from "@/src/contexts/StateContext";

dayjs.extend(localizedFormat);

const HOURS = Array.from({ length: 11 }, (_, i) => `${8 + i}:00`);


interface AppointmentTimePickerProps {
    onConfirmReservation: (date: string, time: string) => void;
    reservedAppointments: ReservedAppointment[];
    testID: string;
}

export default function AppointmentTimePicker({
                                                  onConfirmReservation,
                                                  reservedAppointments,
                                              }: AppointmentTimePickerProps) {
    const [selectedDate, setSelectedDate] = useState(dayjs());
    const [selectedTime, setSelectedTime] = useState<string | null>(null);
    const [showConfirm, setShowConfirm] = useState(false);
    const [showDatePicker, setShowDatePicker] = useState(false);
    const { langId } = useStateContext();

    const locale = useMemo(() => {
        if (langId === 'pl') return 'pl';
        if (langId === 'en') return 'en';
        return 'en';
    }, [langId]);

    const formattedDate = useMemo(() => {
        return dayjs(selectedDate).locale(locale).format('DD MMMM YYYY');
    }, [selectedDate, locale]);

    /**
     * Handles the selection of a specific hour slot.
     * Sets the selected time and triggers the confirmation modal.
     *
     * @param hour - Time slot selected by the user (e.g., "10:00").
     */
    const handleSelectTime = (hour: string) => {
        setSelectedTime(hour);
        setShowConfirm(true);
    };

    /**
     * Confirms the appointment reservation.
     * Triggers the parent callback with selected date and time, and hides the modal.
     */
    const handleConfirm = () => {
        if (selectedTime) {
            onConfirmReservation(formattedDate, selectedTime);
            setShowConfirm(false);
        }
    };

    /**
     * Checks if a specific hour on the currently selected date is already reserved.
     *
     * @param hour - Time slot to check (e.g., "14:00").
     * @returns True if the hour is already reserved for the selected date.
     */
    const isTimeReserved = (hour: string) => {
        return reservedAppointments.some(
            ra => ra.date === formattedDate && ra.time === hour
        );
    };


    /**
     * Callback for the native date picker.
     * Updates the selected date if a new one is picked and closes the picker.
     *
     * @param event - The event triggered by the DateTimePicker.
     * @param selected - The selected Date object, if any.
     */
    const onChangeDate = (event: any, selected?: Date) => {
        setShowDatePicker(false);
        if (selected) {
            setSelectedDate(dayjs(selected));
        }
    };

    return (
        <View className="bg-white p-4">
            <View className="flex-row items-center justify-between mb-4">
                <Pressable onPress={() => setSelectedDate(prev => prev.subtract(1, 'day'))}>
                    <Text className="text-lg">{'←'}</Text>
                </Pressable>
                <Pressable testID={"open-date-picker"} onPress={() => setShowDatePicker(true)}>
                    <Text className="text-base font-medium underline">
                        {formattedDate}
                    </Text>
                </Pressable>
                <Pressable onPress={() => setSelectedDate(prev => prev.add(1, 'day'))}>
                    <Text className="text-lg">{'→'}</Text>
                </Pressable>
            </View>

            {showDatePicker && (
                <View testID="RNDateTimePicker">
                    <DateTimePicker
                        mode="date"
                        display={Platform.OS === 'ios' ? 'spinner' : 'calendar'}
                        value={selectedDate.toDate()}
                        onChange={onChangeDate}
                    />
                </View>
            )}

            <ScrollView horizontal showsHorizontalScrollIndicator={false}>
                {HOURS.map(hour => {
                    const isSelected = hour === selectedTime;
                    const isReserved = isTimeReserved(hour);

                    return (
                        <Pressable
                            key={hour}
                            disabled={isReserved}
                            onPress={() => handleSelectTime(hour)}
                            className={`px-4 py-2 rounded-lg border mx-1 ${
                                isReserved
                                    ? 'bg-gray-300'
                                    : isSelected
                                        ? 'bg-cyan-600'
                                        : 'bg-white'
                            }`}
                            style={{
                                borderColor: isSelected ? '#0e7490' : '#ccc',
                            }}
                            testID={`time-button-${hour}`}
                        >
                            <Text
                                className={`text-sm ${
                                    isReserved
                                        ? 'text-gray-500'
                                        : isSelected
                                            ? 'text-white font-bold'
                                            : 'text-gray-700'
                                }`}
                            >
                                {hour}
                            </Text>
                        </Pressable>
                    );
                })}
            </ScrollView>

            <Modal
                isVisible={showConfirm}
                onBackdropPress={() => setShowConfirm(false)}
                backdropOpacity={0.4}
                animationIn="zoomIn"
                animationOut="zoomOut"
                backdropTransitionOutTiming={0}
                testID={"confirm-modal"}
            >
                <View className="bg-white p-6 rounded-lg items-center">
                    <Text className="text-base mb-4 text-center">
                        {translation[langId].appointmentTimePicker.confirmText}{" "}
                        <Text className="font-semibold">
                            {formattedDate} o {selectedTime}?
                        </Text>
                    </Text>

                    <View className="flex-row gap-6">
                        <Pressable testID={"cancel-reservation"} onPress={() => setShowConfirm(false)}>
                            <Text className="text-red-500 font-medium">{translation[langId].appointmentTimePicker.cancelButtonText}</Text>
                        </Pressable>
                        <Pressable onPress={handleConfirm}>
                            <Text className="text-blue-600 font-semibold">{translation[langId].appointmentTimePicker.confirmButtonText}</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
        </View>
    );
}