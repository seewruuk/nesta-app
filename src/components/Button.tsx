import {Pressable, Text} from "react-native";


interface ButtonProps {
    text: string;
    onPress?: () => void;
}

export default function Button({text, onPress} : ButtonProps){
    return (
        <Pressable
            onPress={onPress}
            className="bg-primary items-center px-4 font-semibold py-4 rounded-xl"
        >
            <Text>{text}</Text>
        </Pressable>
    );
}