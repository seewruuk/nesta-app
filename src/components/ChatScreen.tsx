import React, { useState, useEffect } from 'react';
import { View, Text, FlatList, TextInput, TouchableOpacity } from 'react-native';
import { useRouter } from 'expo-router';
import { useStateContext } from '@/src/contexts/StateContext';
import { Message as MsgType } from '@/src/types/Message';

export default function ChatScreen() {
    const router = useRouter();
    const {
        state: { currentUserId, messages },
        addMessage,
        deleteMessage,
        markAsRead,
    } = useStateContext();

    const [text, setText] = useState('');

    const chatMsgs = [...messages].sort(
        (a, b) => new Date(a.date).getTime() - new Date(b.date).getTime()
    );

    useEffect(() => {
        if (!currentUserId) {
            router.replace('/login');
        }
    }, [currentUserId]);

    useEffect(() => {
        if (!currentUserId) return;
        chatMsgs.forEach(msg => {
            if (msg.receiverId === currentUserId && !msg.isRead) {
                markAsRead(msg.id);
            }
        });
    }, [chatMsgs, currentUserId]);

    if (!currentUserId) return null;

    const handleSend = () => {
        if (!text.trim()) return;
        const newMsg: MsgType = {
            id: `msg${Date.now()}`,
            senderId: currentUserId,
            receiverId: '', // unused in static view
            message: text.trim(),
            date: new Date().toISOString(),
            isRead: false,
            status: 'sent',
        };
        addMessage(newMsg);
        setText('');

        setTimeout(() => {
            const reply: MsgType = {
                id: `bot${Date.now()}`,
                senderId: 'bot',
                receiverId: '',
                message: 'tak',
                date: new Date().toISOString(),
                isRead: true,
                status: 'read',
            };
            addMessage(reply);
        }, 5000);
    };

    const renderItem = ({ item }: { item: MsgType }) => {
        const isMine = item.senderId === currentUserId;
        const isBot = item.senderId === 'bot';
        return (
            <View
                style={{
                    alignSelf: isMine ? 'flex-end' : 'flex-start',
                    backgroundColor: isMine
                        ? '#DCF8C5'
                        : isBot
                            ? '#FFE0B2'
                            : '#FFF',
                    margin: 4,
                    padding: 8,
                    borderRadius: 8,
                    borderWidth: isBot ? 1 : 0,
                    borderColor: isBot ? '#FF9800' : 'transparent',
                }}
            >
                <Text style={isBot ? { fontStyle: 'italic' } : {}}>{item.message}</Text>
                <Text style={{ fontSize: 10, textAlign: 'right' }}>
                    {new Date(item.date).toLocaleTimeString()}
                </Text>
                <TouchableOpacity onPress={() => deleteMessage(item.id)}>
                    <Text style={{ fontSize: 10, color: 'red' }}>Usuń</Text>
                </TouchableOpacity>
            </View>
        );
    };

    return (
        <View style={{ flex: 1, padding: 8, paddingBottom: 150, paddingTop: 50 }}>
            <FlatList
                data={chatMsgs}
                keyExtractor={item => item.id}
                renderItem={renderItem}
            />
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
                <TextInput
                    style={{ flex: 1, borderWidth: 1, borderRadius: 4, padding: 8 }}
                    placeholder="Napisz wiadomość..."
                    value={text}
                    onChangeText={setText}
                />
                <TouchableOpacity onPress={handleSend} style={{ marginLeft: 8 }}>
                    <Text>Wyślij</Text>
                </TouchableOpacity>
            </View>
        </View>
    );
}
