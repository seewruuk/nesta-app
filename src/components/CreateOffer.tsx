import { Text, TextInput, ScrollView, TouchableOpacity, Alert } from 'react-native';
import { useState} from 'react';
import { Offer } from '@/src/data/offers';
import { useRouter } from 'expo-router';
import { useStateContext } from "@/src/contexts/StateContext";
import uuid from 'react-native-uuid';

export default function CreateOffer() {
    const {
        state: { currentUserId }, addOffer
    } = useStateContext();
    const router = useRouter();
    const [title, setTitle] = useState('');
    const [description, setDescription] = useState('');
    const [rentPrice, setRentPrice] = useState('');
    const [availableFrom, setAvailableFrom] = useState('');

    const handleSubmit = () => {
        if (!title || !description || !rentPrice || !availableFrom) {
            Alert.alert('Uzupełnij wszystkie pola!');
            return;
        }

        const newOffer: Offer = {
            id: uuid.v4().toString(),
            apartmentId: 'apt1',
            authorId: currentUserId || 'guest',
            title,
            description,
            rentPrice: parseInt(rentPrice),
            deposit: parseInt(rentPrice),
            extraFees: {
                internet: 'Dodatkowe',
                utilities: 'Dodatkowe',
            },
            availableFrom,
            minTermMonths: 6,
            shortTermAllowed: false,
            preferredTenants: [],
            parkingIncluded: false,
            elevator: false,
            wheelchairAccess: false,
            petsAllowed: 'Do ustalenia',
            tvInternet: false,
            heatingType: 'centralne',
            smokingAllowed: 'Do ustalenia',
            cellar: false,
            status: 'Szkic',
        };

        addOffer(newOffer);
        Alert.alert('Oferta zapisana jako szkic!');
        router.back();
    };

    return (
        <ScrollView className="flex-1 p-4 bg-white">
            <Text className="text-xl font-bold mb-2">Nowa oferta</Text>

            <TextInput className="border rounded p-2 mb-3" placeholder="Tytuł" value={title} onChangeText={setTitle} />
            <TextInput className="border rounded p-2 mb-3" placeholder="Opis" value={description} onChangeText={setDescription} multiline />
            <TextInput className="border rounded p-2 mb-3" placeholder="Cena najmu (PLN)" value={rentPrice} onChangeText={setRentPrice} keyboardType="numeric" />
            <TextInput className="border rounded p-2 mb-3" placeholder="Dostępne od (rrrr-mm-dd)" value={availableFrom} onChangeText={setAvailableFrom} />

            <TouchableOpacity className="bg-blue-600 rounded p-3" onPress={handleSubmit}>
                <Text className="text-white text-center">Zapisz ofertę</Text>
            </TouchableOpacity>
        </ScrollView>
    );
}
