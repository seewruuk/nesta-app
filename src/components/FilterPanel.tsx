/**
 * Filter panel for real estate offers.
 *
 * A slide-in panel component that allows users to input
 * various search filters (e.g., location, price, number of rooms).
 * Appears from the right side when `visible` is true.
 *
 * Filter values are stored in the external `filters` state
 * and updated dynamically via `setFilters`.
 *
 * @param visible - Whether the panel is visible (slide-in).
 * @param onClose - Function called to close the panel.
 * @param filters - Object representing current filter values.
 * @param setFilters - Function to update the filters state.
 */

import { useEffect, useRef } from 'react';
import {
    Animated,
    Pressable,
    Text, TextInput,
    TouchableWithoutFeedback,
    View
} from 'react-native';
import {useStateContext} from "@/src/contexts/StateContext";
import {translation} from "@/src/translation";

export interface Filters {
    city: string;
    priceMin: string;
    priceMax: string;
    roomsMin: string;
    roomsMax: string;
    bedroomsMin: string;
    bedroomsMax: string;
    furnished: string;         
    petsAllowed: string;       
    shortTermAllowed: string;  
}

interface FilterPanelProps {
    visible: boolean;
    onClose: () => void;
    filters: Filters;
    setFilters: (f: Filters) => void;
}

export default function FilterPanel({
                                        visible, onClose, filters, setFilters
                                    }: FilterPanelProps) {
    const slideAnim = useRef(new Animated.Value(0)).current;
    const { langId } = useStateContext();


    useEffect(() => {
        Animated.timing(slideAnim, {
            toValue: visible ? 1 : 0,
            duration: 300,
            useNativeDriver: true
        }).start();
    }, [visible]);

    const translateX = slideAnim.interpolate({
        inputRange: [0, 1],
        outputRange: [400, 0]
    });

    return (
        <>
            {visible && (
                <TouchableWithoutFeedback onPress={onClose}>
                    <View className="absolute inset-0 bg-black/80" />
                </TouchableWithoutFeedback>
            )}
            <Animated.View style={{
                transform: [{ translateX }],
                position: 'absolute', right: 0, top: 0, bottom: 0,
                width: '80%', backgroundColor: 'white', padding: 20, zIndex: 1000,
                paddingTop: 80
            }}>
                <Text className="text-lg font-bold mb-4">{translation[langId].filterPanel.filterOffers}</Text>

                <Text className="font-semibold">{translation[langId].filterPanel.city}</Text>
                <TextInput
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.cityPlaceholder}
                    value={filters.city}
                    onChangeText={city => setFilters({ ...filters, city })}
                />

                <Text className="font-semibold">{translation[langId].filterPanel.priceMin}</Text>
                <TextInput
                    keyboardType="numeric"
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.priceMinPlaceholder}
                    value={filters.priceMin}
                    onChangeText={priceMin => setFilters({ ...filters, priceMin })}
                />
                <Text className="font-semibold">{translation[langId].filterPanel.priceMax}</Text>
                <TextInput
                    keyboardType="numeric"
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.priceMaxPlaceholder}
                    value={filters.priceMax}
                    onChangeText={priceMax => setFilters({ ...filters, priceMax })}
                />

                <Text className="font-semibold">{translation[langId].filterPanel.roomsMin}</Text>
                <TextInput
                    keyboardType="numeric"
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.roomsMinPlaceholder}
                    value={filters.roomsMin}
                    onChangeText={roomsMin => setFilters({ ...filters, roomsMin })}
                />
                <Text className="font-semibold">{translation[langId].filterPanel.roomsMax}</Text>
                <TextInput
                    keyboardType="numeric"
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.roomsMaxPlaceholder}
                    value={filters.roomsMax}
                    onChangeText={roomsMax => setFilters({ ...filters, roomsMax })}
                />

                <Text className="font-semibold">{translation[langId].filterPanel.bedroomsMin}</Text>
                <TextInput
                    keyboardType="numeric"
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.bedroomsMinPlaceholder}
                    value={filters.bedroomsMin}
                    onChangeText={bedroomsMin => setFilters({ ...filters, bedroomsMin })}
                />
                <Text className="font-semibold">{translation[langId].filterPanel.bedroomsMax}</Text>
                <TextInput
                    keyboardType="numeric"
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.bedroomsMaxPlaceholder}
                    value={filters.bedroomsMax}
                    onChangeText={bedroomsMax => setFilters({ ...filters, bedroomsMax })}
                />

                <Text className="font-semibold">{translation[langId].filterPanel.furnished}</Text>
                <TextInput
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.furnishedPlaceholder}
                    value={filters.furnished}
                    onChangeText={furnished => setFilters({ ...filters, furnished })}
                />

                <Text className="font-semibold">{translation[langId].filterPanel.animals}</Text>
                <TextInput
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.animalsPlaceholder}
                    value={filters.petsAllowed}
                    onChangeText={petsAllowed => setFilters({ ...filters, petsAllowed })}
                />

                <Text className="font-semibold">{translation[langId].filterPanel.short}</Text>
                <TextInput
                    className="border border-gray-400 rounded p-2 mb-4"
                    placeholder={translation[langId].filterPanel.shortPlaceholder}
                    value={filters.shortTermAllowed}
                    onChangeText={shortTermAllowed => setFilters({ ...filters, shortTermAllowed })}
                />

                <Pressable
                    onPress={onClose}
                    className="mt-4 bg-gray-200 rounded p-3 items-center"
                >
                    <Text>{translation[langId].filterPanel.close}</Text>
                </Pressable>
            </Animated.View>
        </>
    );
}
