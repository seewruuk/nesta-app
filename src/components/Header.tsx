import { Text } from 'react-native'
import React from 'react'


interface HeaderProps {
    text: string,
    className?: string
}

export default function Header({ text, className }: HeaderProps) {

    const defaultClassName = "text-[20px] font-bold text-black leading-snug";

    return <Text className={`${defaultClassName} ${className || ''}`}>{text}</Text>
}
