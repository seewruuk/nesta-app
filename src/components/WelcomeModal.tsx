// src/components/WelcomeModal.tsx
import React, { useState } from 'react'
import { View, Text, TextInput, Pressable, ScrollView } from 'react-native'
import { useRouter } from 'expo-router'

const FLAT_TYPES = [
    { label: 'Kawalerka', value: 'studio' },
    { label: '2 pokoje', value: '2' },
    { label: '3+ pokoje', value: '3+' },
]

export const WelcomeModal: React.FC = () => {
    const router = useRouter()
    const [mode, setMode] = useState<'flatmate' | 'rent'>('rent')
    const [city, setCity] = useState('')
    const [minPrice, setMinPrice] = useState('')
    const [maxPrice, setMaxPrice] = useState('')
    const [type, setType] = useState('')
    const [dropdownOpen, setDropdownOpen] = useState(false)

    const onSearch = () => {
        if (mode === 'flatmate') {
            router.push('/posts')
        } else {
            const params = new URLSearchParams({ city, minPrice, maxPrice, type })
            router.push(`/offers?${params.toString()}`)
        }
    }

    const selectedTypeLabel = FLAT_TYPES.find(t => t.value === type)?.label

    return (
        <View className="absolute inset-0 justify-center items-center bg-black bg-opacity-50">
            <View className="w-10/12 bg-white rounded-lg p-6 space-y-4">
                {/* Segmentacja */}
                <View className="flex-row justify-around">
                    <Pressable onPress={() => setMode('flatmate')}>
                        <Text className={`text-lg ${mode === 'flatmate' ? 'font-bold text-black' : 'text-gray-500'}`}>
                            Znajdź współlokatora
                        </Text>
                    </Pressable>
                    <Pressable onPress={() => setMode('rent')}>
                        <Text className={`text-lg ${mode === 'rent' ? 'font-bold text-black' : 'text-gray-500'}`}>
                            Wynajem
                        </Text>
                    </Pressable>
                </View>

                {/* Miasto */}
                <TextInput
                    value={city}
                    onChangeText={setCity}
                    placeholder="Miasto"
                    className="border border-gray-300 rounded px-3 py-2"
                />

                {/* Tylko dla Wynajem */}
                {mode === 'rent' && (
                    <>
                        {/* Zakres ceny */}
                        <View className="flex-row space-x-2">
                            <TextInput
                                value={minPrice}
                                onChangeText={setMinPrice}
                                placeholder="Min cena"
                                keyboardType="numeric"
                                className="flex-1 border border-gray-300 rounded px-3 py-2"
                            />
                            <TextInput
                                value={maxPrice}
                                onChangeText={setMaxPrice}
                                placeholder="Max cena"
                                keyboardType="numeric"
                                className="flex-1 border border-gray-300 rounded px-3 py-2"
                            />
                        </View>

                        {/* Custom Dropdown */}
                        <View className="relative">
                            <Pressable
                                className="border border-gray-300 rounded px-3 py-2 flex-row justify-between items-center"
                                onPress={() => setDropdownOpen(prev => !prev)}
                            >
                                <Text className={`${selectedTypeLabel ? 'text-black' : 'text-gray-400'}`}>
                                    {selectedTypeLabel || 'Typ mieszkania'}
                                </Text>
                                <Text className="text-gray-500">{dropdownOpen ? '▲' : '▼'}</Text>
                            </Pressable>
                            {dropdownOpen && (
                                <ScrollView
                                    className="absolute top-full mt-1 w-full max-h-40 bg-white border border-gray-300 rounded"
                                >
                                    {FLAT_TYPES.map(option => (
                                        <Pressable
                                            key={option.value}
                                            onPress={() => {
                                                setType(option.value)
                                                setDropdownOpen(false)
                                            }}
                                            className="px-3 py-2"
                                        >
                                            <Text className="text-black">{option.label}</Text>
                                        </Pressable>
                                    ))}
                                </ScrollView>
                            )}
                        </View>
                    </>
                )}

                {/* Przycisk Szukaj */}
                <Pressable
                    onPress={onSearch}
                    className="bg-primary rounded-lg py-3 items-center mt-2"
                >
                    <Text className="text-black text-lg font-medium">Szukaj</Text>
                </Pressable>
            </View>
        </View>
    )
}
