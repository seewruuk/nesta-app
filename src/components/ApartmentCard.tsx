import React from 'react'
import {View, Text, Image, Pressable} from 'react-native'
import {Apartment} from '@/src/types'
import Button from "./Button"
import Header from "./Header"
import {useRouter} from "expo-router";

interface ApartmentCardProps {
    apartment: Apartment
}

export default function ApartmentCard({apartment}: ApartmentCardProps) {

    const router = useRouter();

    return (
        <View className="bg-white p-4 rounded-xl shadow-md mb-4">
            <Pressable
                onPress={() => router.push(`/offers/${apartment.id}`)}
                className="h-[200px] w-full rounded-xl overflow-hidden mb-4 "
            >
                <Image
                    source={apartment.images[0]}
                    className="object-cover w-full"
                    resizeMode="cover"
                />
            </Pressable>

            <View className={"flex flex-col gap-3"}>
                <Text className={"text-[11px] text-gray-500"}>{apartment.city}</Text>
                <Header text={apartment.title} className={"text-[15px]"}/>
                <Text>{apartment.price}PLN / msc.</Text>
                <View>
                    <View className={"flex flex-row gap-2"}>
                        <Text className={"text-[#C5C5C5] text-[11px]"}>Powierzchnia</Text>
                        <Text className={"text-[#C5C5C5] text-[11px]"}>{apartment.area} m2</Text>
                    </View>
                </View>

                <View className={"flex flex-row gap-4 justify-between items-center"}>

                    <Pressable
                        onPress={() => router.push(`/offers/${apartment.id}`)}
                        className="bg-primary py-4 flex-1  rounded-xl"
                    >
                      <Text className={"text-center"}>Zobacz wiÄ™cej</Text>
                    </Pressable>

                    <Pressable
                        onPress={() => router.push(`/`)}
                        className="bg-black py-4 flex-1 rounded-xl"
                    >
                        <Text className={"text-white text-center"}>Kontakt</Text>
                    </Pressable>
                </View>
            </View>
        </View>

    )
}
