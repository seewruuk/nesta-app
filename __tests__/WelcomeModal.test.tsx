import React from 'react';
import { render } from '@testing-library/react-native';
import { WelcomeModal } from '../src/components/WelcomeModal';

describe('<WelcomeModal />', () => {


    it('powinien się zgadzać ze snapshotem', () => {
        const tree = render(<WelcomeModal />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renderuje wszystkie pola i przyciski', () => {
        const { getByPlaceholderText, getByText } = render(<WelcomeModal />);

        // inputs
        expect(getByPlaceholderText('Miasto')).toBeTruthy();
        expect(getByPlaceholderText('Min. cena')).toBeTruthy();
        expect(getByPlaceholderText('Max. cena')).toBeTruthy();

        // przyciski Pressable
        expect(getByText('Wynajem')).toBeTruthy();
        expect(getByText('Znd. Współlokatora')).toBeTruthy();

        // główny button Szukaj
        expect(getByText('Szukaj')).toBeTruthy();
    });
});
