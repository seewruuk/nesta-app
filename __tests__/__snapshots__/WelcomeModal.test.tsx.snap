// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeModal /> powinien się zgadzać ze snapshotem 1`] = `
<View
  className="flex-1 justify-center items-center relative"
>
  <View
    collapsable={false}
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": [
          {
            "translateX": 0,
          },
        ],
        "width": 750,
        "zIndex": -2,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "testUri": "../../../src/assets/images/background-image.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.96)",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 750,
        "zIndex": -1,
      }
    }
  />
  <View
    className="w-10/12 bg-white rounded-3xl p-[32px] space-y-4 flex flex-col gap-8 z-10"
  >
    <View>
      <Text
        className="text-[20px] font-bold text-black leading-snug py-4 text-3xl"
      >
        Witaj ponownie!
      </Text>
      <Text
        className="text-[15px] leading-7 text-[#1C2635] "
      >
        U nas wymarzone nieruchomości znajdują właścicieli swoich marzeń!
      </Text>
    </View>
    <View
      className="flex flex-col"
    >
      <View
        className="flex flex-row justify-between"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="border-t border-r border-l border-gray-300 flex-1 rounded-tr-2xl rounded-tl-2xl flex items-center justify-center py-4 font-semibold"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text>
            Wynajem
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="border border-gray-300 bg-[#F5F5F5] flex-1 -ml-4 rounded-tr-2xl flex items-center justify-center py-4 "
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text>
            Znd. Współlokatora
          </Text>
        </View>
      </View>
      <View
        className="flex gap-2 px-4 border-b border-r border-l border-gray-300 rounded-br-2xl rounded-bl-2xl z-50 bg-white py-5"
      >
        <TextInput
          className="bg-gray-100 rounded-lg px-3 py-3"
          onChangeText={[Function]}
          placeholder="Miasto"
          value=""
        />
        <View
          className="flex flex-row gap-2"
        >
          <TextInput
            className="bg-gray-100 rounded-lg px-3 py-3 flex-1"
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Min. cena"
            value=""
          />
          <TextInput
            className="bg-gray-100 rounded-lg px-3 py-3 flex-1"
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Max. cena"
            value=""
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="bg-primary items-center px-4 font-semibold py-4 rounded-xl"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text>
            Szukaj
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
